eth/vm/base.py

```diff
@@ -333,6 +372,22 @@ def import_block(self, block: BlockAPI) -> BlockAndMetaWitness:
             receipts,
         )
 
+        if hasattr(block, "withdrawals") and block.withdrawals not in (None, ()):
```

Comment:
**Note for refactoring:** This was made to work, re-think this logic a bit. Not just here, but this sort of logic appears in a few other places.